Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -237,6 +283,22 @@ public class XPCServer {
         self.routes[route.route] = ConstrainedXPCHandlerWithMessageWithReply(handler: handler)
     }
     
+    internal func addConnection(_ connection: xpc_connection_t) {
+        // Keep a weak reference to the connection and this server, setting this as the context on the connection
+        let weakConnection = WeakConnection(connection, server: self)
+        self.connections.insert(weakConnection)
+        xpc_connection_set_context(connection, Unmanaged.passUnretained(weakConnection).toOpaque())
+        
+        // The finalizer is called when the connection's retain count has reached zero, so now we need to remove the
+        // wrapper from the containing connections array
+        xpc_connection_set_finalizer_f(connection, { opaqueWeakConnection in
+            if let opaqueWeakConnection = opaqueWeakConnection {
```

Comment:
When can the `opaqueWeeakConnection` be `nil`? As far as I can tell, that should never happen, and this should be made fail loudly (`fatalError`)

Reply:
It shouldn't ever be, updated.